### PR TITLE
ipatests: Test for IPATrustControllerPrincipalCheck

### PR DESCRIPTION
**Pull request by menonsudhir**
This testcase checks when trust between IPA-AD is established successfully, 
IPATrustControllerPrincipalCheck displays result as SUCCESS

### CODE REVIEW

**Comment by rcritten** [comment]
This check verifies that the cifs principal is a member of the group cn=adtrust agents. This test should also remove the principal from that entry and re-run to ensure it is properly caught.

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 1]
```diff
@@ -1399,6 +1399,53 @@ def test_ipahealthcheck_trustcontoller_conf_check(self):
...
+        entry['member'] = krbprinc
+        ldap.update_entry(entry)  # pylint: disable=no-member
+
+    def test_trustcontroller_principalcheck(self, modify_cifs_princ):
+        """
+        This testcase checks when trust between IPA-AD is established
+        without any errors, IPATrustControllerPrincipalCheck displays
+        result as ERROR and when cifs principal is removed
+        """
+        error_msg = "{key} is not a member of {group}"
```
Comment (flo-renaud) [inline ipatests/test_integration/test_ipahealthcheck.py:1433]: There should be 2 tests: 
1- run the check when everything is setup properly, ensure IPATrustControllerPrincipalCheck returns success
2- remove the cifs/hostname@realm principal from the adtrust agents, run the check, ensure IPATrustControllerPrincipalCheck  detects the issue

**Review by flo-renaud — COMMENTED** [review]
Hi @menonsudhir 
please find inline comments

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 2]
```diff
@@ -1408,6 +1408,72 @@ def test_ipahealthcheck_trustcontoller_conf_check(self):
...
+        keyname = "cifs/"'{}'.format(self.master.hostname) + "@"'{}'.format(
+            self.master.domain.realm
+        )
```
Comment (flo-renaud) [inline ipatests/test_integration/test_ipahealthcheck.py:1445]: You can simplify with
`keyname = "cifs/{}@{}".format(self.master.hostname, self.master.domain.realm)`

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 3]
```diff
@@ -1408,6 +1408,72 @@ def test_ipahealthcheck_trustcontoller_conf_check(self):
...
+
+    def test_principalcheck_with_cifs_entry(self):
+        """
+        This testcase checks IPATrustControllerPrincipalCheck
+        displays result as SUCCESS when cifs principal is present
+        in cn=adtrust agents group
+        """
+        keyname = "cifs/"'{}'.format(self.master.hostname) + "@"'{}'.format(
+            self.master.domain.realm
+        )
```
Comment (flo-renaud) [inline ipatests/test_integration/test_ipahealthcheck.py:1466]: Same comment as above, you can simplify with
keyname = "cifs/{}@{}".format(self.master.hostname, self.master.domain.realm)

**Review by flo-renaud — COMMENTED** [review]
Hi @menonsudhir 
please find inline comments

File: ipatests/test_integration/test_ipahealthcheck.py (hunk 3 above)
Comment (menonsudhir) [reply ipatests/test_integration/test_ipahealthcheck.py:1466]: @flo-renaud have made the changes as mentioned.

**Comment by flo-renaud** [comment]
For future ref, the temp test result is available [here](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/55d300e6-4036-11eb-927b-fa163ec3b944/) and was green.

**Comment by flo-renaud** [comment]
master:

* d7eece2ea1d7e629cd87371df18143bfde7ae365 ipatests: Test for IPATrustControllerPrincipalCheck

**Pull request closed without merging**